cmd/libvirt-actuator/main.go

```diff
@@ -109,7 +133,244 @@ func init() {
 			}
 			return nil
 		},
+	}
+}
+
+func BuildPKSecret(secretName, namespace, pkLoc string) (*apiv1.Secret, error) {
+	pkBytes, err := ioutil.ReadFile(pkLoc)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read %v: %v", pkLoc, err)
+	}
+
+	return &apiv1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			"privatekey": pkBytes,
+		},
+	}, nil
+}
+
+func createSecretAndWait(f *framework.Framework, secret *apiv1.Secret) error {
+	_, err := f.KubeClient.CoreV1().Secrets(secret.Namespace).Create(secret)
+	if err != nil {
+		return err
+	}
+
+	err = wait.Poll(framework.PollInterval, framework.PoolTimeout, func() (bool, error) {
+		if _, err := f.KubeClient.CoreV1().Secrets(secret.Namespace).Get(secret.Name, metav1.GetOptions{}); err != nil {
+			return false, nil
```

Comment:
This, and the following `return false, nil`, can be collapsed to:

    return err != nil, err